pyiron_base/job/template.py

```diff
@@ -24,39 +24,57 @@
 class TemplateJob(GenericJob):
     def __init__(self, project, job_name):
         super().__init__(project, job_name)
-        self.input = GenericParameters(table_name="input")
+        self._input = InputList(table_name="input")
+
+    @property
+    def input(self):
+        return self._input
+
+    @input.setter
+    def input(self, input):
+        self._input = InputList(input, table_name="input")
 
     def to_hdf(self, hdf=None, group_name=None):
         super().to_hdf(
             hdf=hdf,
             group_name=group_name
         )
-        with self.project_hdf5.open("input") as h5in:
-            self.input.to_hdf(h5in)
+        if hdf is None:
+            hdf = self.project_hdf5
+        self._input.to_hdf(hdf=hdf, group_name=None)
 
     def from_hdf(self, hdf=None, group_name=None):
         super().from_hdf(
             hdf=hdf,
             group_name=group_name
         )
-        with self.project_hdf5.open("input") as h5in:
-            self.input.from_hdf(h5in)
+        if hdf is None:
```

Comment:
This part should be handled in the `super()` call.

Reply:
So, I should be able to just use `self._hdf5` , since in the super call, this is set to hdf if `hdf is not None`. Therefore, it has to be impossible to call this when `self._hdf` is not set - is this guaranteed to be always be the case?

Reply:
Yes, you should be fine using just `self._hdf5`.